### PR TITLE
Support label_smoothing in cross_entropy

### PR DESCRIPTION
**Pull request by jbschlosser**
Adds awareness of the new `label_smoothing` functionality in `cross_entropy` added in https://github.com/pytorch/pytorch/pull/63122 to nestedtensor's implementation.

Fixes broken test: `test.test_nested_tensor_functional.TestFunctional`.

### CODE REVIEW

File: nestedtensor/csrc/python_functions.cpp [hunk 1]
```diff
@@ -32,6 +33,9 @@ at::Tensor cross_entropy(
   if (ignore_index.has_value()) {
     options = options.ignore_index(ignore_index.value());
   }
+  if (label_smoothing.has_value()) {
+    options = options.label_smoothing(label_smoothing.value());
```
Comment (jbschlosser) [inline nestedtensor/csrc/python_functions.cpp:37]: Check failures are due to `F::CrossEntropyFuncOptions` not having a `label_smoothing` arg, although it's present in master (https://github.com/pytorch/pytorch/blob/92b31b59aff908fcecb76e5baff5bb52ce62608a/torch/csrc/api/include/torch/nn/options/loss.h#L666). What's the deal here?

**Comment by jbschlosser** [comment]
Closing and reopening diff from the FB side
